src/utils/validateProperties.ts

```diff
@@ -0,0 +1,14 @@
+import { PineconeArgumentError } from '../errors';
+
+export const ValidateProperties = (item: any, validProperties: any[]) => {
```

Comment:
See my comment above about this: I think we need to do something like this:

```suggestion
function validateProperties<T>(item: T, validProperties: keysOf<T>)
```

Basically, we want to couple the type to the list of validation keys being passed in.

Reply:
[update: we spoke offline and decided to simplify things -- we are just making custom constants to hold the property values & check against those]